### ISSUE TITLE
Keep review approvals

### ISSUE THREAD

**Issue by sschuberth** [cirrus-actions/rebase#6]
I think it would be nice if this rebase action was able to keep review approvals even if the project's settings have "Dismiss stale pull request approvals when new commits are pushed" enabled. The rationale is that a rebase does not change the contents of the PR (unless conflicts need to be resolved, but then this action does not work anyway), so if it was approved before the rebase, the same contents should also be approved afterwards. We *do* want to re-run CI, however, as the rebase might introduce changes that break the PR's functionality.

**Comment by fkorotkov** [comment]
I'm not sure if actions have enough permissions to reinstate the approvals from other users. 🤔

**Comment by stefan-enway** [comment]
Approvals are kept.

**Comment by sschuberth** [comment]
@stefan-enway, no, not if "Dismiss stale pull request approvals when new commits are pushed" is enabled, as stated above.

**Comment by ivandimitrovSC** [comment]
I would like to make an action for that, as it is not part of actually of the rebase. So if my manager decides that it is worth spending my time on that, I will write one and link it here as well.

**Comment by sschuberth** [comment]
@ivandimitrovSC under what circumstances *other* than rebase would you want to keep approvals? What I'm trying to say is: Keeping approvals is tightly related to the rebase action which would otherwise dismiss approvals (if the above mentioned option is checked). As such, keeping approvals should be implemented as part of this action, and not of another action. Also, this should be configurable: Some user might actually *want* the rebase to dismiss approvals.

**Comment by ivandimitrovSC** [comment]
@sschuberth I would just make a reusable piece of code. My point is that approvals should be reapplied _only_ when rebasing automatically. Still, this is not a core functionality to the rebase (git is not GitHub and AFAIK it doesn't have PR requests approval). This might be an option or something else, but IMO it is just nice to have.

**Comment by ivandimitrovSC** [comment]
Well, I won't be taking this task in a Sprint.

**Comment by ZacSweers** [comment]
This is more of a github problem than this action. Github won't remove approvers on clean merge commits, but not rebases
